RtpTools/RtpDump.cs

```diff
@@ -247,107 +247,111 @@ void ReadBinaryFileHeader()
-                    if (entry.Format < FileFormat.Text)
-                    {
-                        //The unexpected data consists of the supposed fileheader.
-                        if (m_FileIdentifier == null)
-                        {
-                            //Assign it
-                            m_FileIdentifier = new (unexpectedData);
 
-                            //Read the following Binary File Header
-                            ReadBinaryFileHeader();
+                      //The unexpected data consists of the supposed fileheader.
+                      if (m_FileIdentifier == null)
+                      {
+                        //Assign it
+                        m_FileIdentifier = new (unexpectedData);
 
-                            //no more data is unexpected.
-                            unexpectedData = null;
 
-                            //Remove the offset which was not related to an entry.
-                            m_Offsets.RemoveAt(offsetsCount - 1);
+                        //Read the following Binary File Header
+                        ReadBinaryFileHeader();
 
-                            //Read the entry.
-                            return ReadToolEntry();
-                        }
-                        else Media.Common.TaggedExceptionExtensions.RaiseTaggedException(unexpectedData, "Encountered a Binary file header when already parsed the header. The Tag property contains the data unexpected.");
+                        //no more data is unexpected.
+                        unexpectedData = null;
+
+                        //Remove the offset which was not related to an entry.
+                        m_Offsets.RemoveAt(offsetsCount - 1);
+
+                        //Read the entry.
+                        return ReadToolEntry();
                     }
-                    else if (unexpectedData != null) Media.Common.TaggedExceptionExtensions.RaiseTaggedException(entry, "Unexpected data found while parsing a Text format. See the Tag property of the InnerException", new Common.TaggedException<Common.MemorySegment>(unexpectedData));
+                    else if (unexpectedData != null) Media.Common.TaggedExceptionExtensions.RaiseTaggedException(unexpectedData, "Encountered a Binary file header when already parsed the header. The Tag property contains the data unexpected.");
                 }                    
 
                 //Call determine format so item has the correct format (Header [or Payload])
                 if (foundFormat == FileFormat.Unknown)
                 {
                     Media.Common.TaggedExceptionExtensions.RaiseTaggedException(entry, "Unknown format");
                 }
+                else if (unexpectedData != null) Media.Common.TaggedExceptionExtensions.RaiseTaggedException(entry, "Unexpected data found while parsing a Text format. See the Tag property of the InnerException", new Common.TaggedException<Common.MemorySegment>(unexpectedData));
```

Comment:
It seems this branch is added here but it was already present just a few lines up, let me know why you think we should move this logic if it was intentional.

Reply:
I've just removed try / catch here